scraper.rb

```diff
@@ -47,11 +47,41 @@ class MemberPage < Scraped::HTML
     url.to_s
   end
 
+  field :start_date do
+    date_from(
+      recent_electoral_history
+                  .scan(/(?:Elected on|Oath of Allegiance|casual elections on)[^:]*\:\s*(\d{1,2}\.\d{1,2}\.\d{2,4})/)
+                  .flatten
+                  .last
+    )
+  end
+
+  field :end_date do
+    date_from(
+      recent_electoral_history
+                  .scan(/(?:Resignation from Parliament)[^:]*\:\s*(\d{1,2}\.\d{1,2}\.\d{2,4})/)
+                  .flatten
+                  .last
+    )
+  end
+
   private
 
   def box
     noko.css('div.column2')
   end
+
+  def recent_electoral_history
```

Comment:
This is a slightly misleading name. If I'm reading the code correctly, it's finding the data about the chronologically-last Parliament that the person was a member of, rather than being to do with recency itself.